### PR TITLE
build: Upgrade DF to 36.0.0 and arrow-rs 50.0.0

### LINKED ISSUE #56: Upgrade DF to 36.0.0, arrow-rs to 50.0.0
### What is the problem the feature request solves?

Upgrade Comet dependencies to most recent DF to 36.0.0, arrow-ts to 50.0.0 to benefit with latest fixes

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #56.

### CODE REVIEW

**Review by viirya — CHANGES_REQUESTED** [review]
Let's hold this until internal patches are ported to Comet. Thanks.

**Comment by viirya** [comment]
Yea

**Comment by comphead** [comment]
> I think we should wait for 36.0.1?

Just curious, what is so interesting in 36.0.1? 🤔

**Comment by viirya** [comment]
I think @andygrove encountered circular dependency issue while releasing 36.0.0. So there is 36.0.1 after fixing the issue.

**Comment by sunchao** [comment]
Looks like 36.0.1 is abandoned. Shall we move forward with this PR?

**Comment by viirya** [comment]
Sounds good to me.

File: core/src/execution/datafusion/expressions/scalar_funcs.rs [hunk 1]
```diff
@@ -366,7 +364,12 @@ fn spark_round(
                 let (precision, scale) = get_precision_scale(data_type);
                 make_decimal_array(array, precision, scale, &f)
             }
-            _ => make_scalar_function(math_expressions::round)(args),
+            DataType::Float32 | DataType::Float64 => {
+                Ok(ColumnarValue::Array(math_expressions::round(&[
+                    array.clone()
+                ])?))
+            }
+            dt => exec_err!("Not supported datatype[{dt}] for ROUND"),
```
Comment (viirya) [inline core/src/execution/datafusion/expressions/scalar_funcs.rs:372]: Hm, why wrap `dt` with `[]`? I don't see there is something in Comet or DataFusion.

Comment (viirya) [inline core/src/execution/datafusion/expressions/scalar_funcs.rs:372]:
```suggestion
            dt => exec_err!("Not supported datatype for ROUND: {dt}"),
```

File: core/src/execution/datafusion/expressions/scalar_funcs.rs [hunk 2]
```diff
@@ -386,7 +389,10 @@ fn spark_round(
                 let (precision, scale) = get_precision_scale(data_type);
                 make_decimal_scalar(a, precision, scale, &f)
             }
-            _ => make_scalar_function(math_expressions::round)(args),
+            ScalarValue::Float32(_) | ScalarValue::Float64(_) => Ok(ColumnarValue::Scalar(
+                ScalarValue::try_from_array(&math_expressions::round(&[a.to_array()?])?, 0)?,
+            )),
+            dt => exec_err!("Not supported datatype[{dt}] for ROUND"),
```
Comment (viirya) [inline core/src/execution/datafusion/expressions/scalar_funcs.rs:395]:
```suggestion
            dt => exec_err!("Not supported datatype for ROUND: {dt}"),
```

**Review by viirya — APPROVED** [review]

**Review by sunchao — APPROVED** [review]

**Pull request merged**